kodilla-testing/src/test/java/com/kodilla/testing/library/BookDirectoryTestSuite.java

```diff
@@ -0,0 +1,136 @@
+
+        // Then
+        assertEquals(4, theListOfBooks.size());
+    }
+
+    @Test
+    public void testListBooksWithConditionMoreThan20() {
+        // Given
+        List<Book> resultListOf0Books = new ArrayList<Book>();
+        List<Book> resultListOf15Books = generateListOfNBooks(15);
+        List<Book> resultListOf40Books = generateListOfNBooks(40);
+        when(libraryDatabaseMock.listBooksWithCondition(anyString()))
+                .thenReturn(resultListOf15Books);
+        when(libraryDatabaseMock.listBooksWithCondition("ZeroBooks"))
+                .thenReturn(resultListOf0Books);
+        when(libraryDatabaseMock.listBooksWithCondition("FortyBooks"))
+                .thenReturn(resultListOf40Books);
+
+        // When
+        List<Book> theListOfBooks0 = bookLibrary.listBooksWithCondition("ZeroBooks");
+        List<Book> theListOfBooks15 = bookLibrary.listBooksWithCondition("Any title");
+        List<Book> theListOfBooks40 = bookLibrary.listBooksWithCondition("FortyBooks");
+        // Then
+
+        assertEquals(0, theListOfBooks0.size());
+        assertEquals(15, theListOfBooks15.size());
+        assertEquals(0, theListOfBooks40.size());
+    }
+
+    @Test
+    public void testListBooksWithConditionFragmentShorterThan3() {
+        // Given
+        List<Book> resultListOf10Books = generateListOfNBooks(10);
+        when(libraryDatabaseMock.listBooksWithCondition(anyString()))
+                .thenReturn(resultListOf10Books);
+
+        // When
+        List<Book> theListOfBooks10 = bookLibrary.listBooksWithCondition("An");
+
+        // Then
+        assertEquals(0, theListOfBooks10.size());
+        verify(libraryDatabaseMock, times(0)).listBooksWithCondition(anyString());
+    }
+
+    @Test
+    public void testListBooksInHandsOfNoRentedBooks() {
+        //Given
+        List<Book> emptyBookList = new ArrayList<>();
+        when(libraryDatabaseMock.listBooksInHandsOf(any(LibraryUser.class))).thenReturn(emptyBookList);
+        //When
+        List<Book> theListOfBooks = bookLibrary.listBooksInHandsOf(any(LibraryUser.class));
+        //Then
+        assertEquals(0, theListOfBooks.size());
+    }
+
+    @Test
+    public void testListBooksInHandsOfOneRentedBook() {
+        //Given
+        List<Book> oneBookList = generateListOfNBooks(1);
+        when(libraryDatabaseMock.listBooksInHandsOf(any(LibraryUser.class))).thenReturn(oneBookList);
+        //When
+        List<Book> theListOfBooks = bookLibrary.listBooksInHandsOf(any(LibraryUser.class));
+        //Then
+        assertEquals(1, theListOfBooks.size());
+    }
+
+    @Test
+    public void testListBooksInHandsOfFiveRentedBooks() {
+        //Given
+        List<Book> fiveBookList = generateListOfNBooks(5);
+        when(libraryDatabaseMock.listBooksInHandsOf(any(LibraryUser.class))).thenReturn(fiveBookList);
+        //When
+        List<Book> theListOfBooks = bookLibrary.listBooksInHandsOf(isA(LibraryUser.class));
```

Comment:
Kombinowałem czy dać tutaj `isA()`, czy `any()`, i zostawiłem niechcący tak jak widać. Nie bardzo rozumiem różnicę, obie wersje działają, ale wiem że to może być pozorne tylko ;)